wallet/loader.go

```diff
@@ -288,6 +292,14 @@ func (l *Loader) UnloadWallet() error {
 	return nil
 }
 
+// SetWaitFn sets a variable function which needs to be waited upon before
+// unloading.
+func (l *Loader) SetWaitFn(fn func()) {
```

Comment:
I don't really care for doing it this way but with the way startup/shutdown is currently implemented, i can't think of a much better way.  #481 should take care of this though.  Can you add a TODO here to reconsider whether this is needed after that issue is resolved?